src/mixins/namex-request-mixin.ts

```diff
@@ -56,4 +57,44 @@ export default class NamexRequestMixin extends Mixins(DateMixin) {
     // Otherwise, the NR is consumable.
     return NameRequestStates.APPROVED
   }
+
+  /**
+   * Generate name request state for the store
+   * @param nr the name request response payload
+   * @param filingId the filing id
+   */
+  generateNameRequestState (nr: any, filingId: number): NameRequestIF {
```

Comment:
I imagine `nr: any` because we don't know fully what to expect from namex at this point, is that right?

Reply:
Yes, There is a defined structure to which the incoming data gets parsed. But there is no structure  defined for incoming data